kfs-prje/src/main/java/edu/arizona/kfs/module/prje/service/impl/PRJEServiceImpl.java

```diff
@@ -0,0 +1,857 @@
+                    ObjectType ot = oc.getFinancialObjectType();
+                    debitEntry.setTransactionDebitCreditCode(deriveDebitSign(ot, balance));
+                }
+                else {
+                    LOG.warn("---- No Object Code " + objectCode + " in " + fiscalYear);
+                }
+            }
+            else {
+                LOG.warn("---- No Object Code -- Can't Derive Debit/Credit Code");
+            }
+            
+            debitEntry.setUniversityFiscalPeriodCode(currentPeriod);
+            debitEntry.setUniversityFiscalYear(fiscalYear);
+            
+            debitEntry.setTransactionDate(transactionDate);
+            debitEntry.setTransactionLedgerEntryAmount(amount.abs());
+            debitEntry.setTransactionLedgerEntrySequenceNumber(lastOriginSequence++);
+            debitEntry.setTransactionScrubberOffsetGenerationIndicator(true);
+
+            String description = generateDebitDescription(transferRecord);
+            debitEntry.setTransactionLedgerEntryDescription(description);
+            
+            return debitEntry;
+        }
+        
+        private OriginEntryFull generateCreditEntry(PRJETransferRecord transferRecord) {
+            PRJEAccountLine accountLine = transferRecord.getAccountLine();
+            KualiDecimal amount = transferRecord.getAmount();
+            KualiDecimal balance = transferRecord.getBalance();
+            
+            LOG.info("--- Credit " + amount + " to Account " + accountLine.getAccountNumber());
+            
+            // If Sub-Account and Sub-Objects exist, then use them. If 
+            // Override accounts and objects do not have sub account or 
+            // sub object, then post them at the override account and 
+            // object.
+            
+            PRJEType type = accountLine.getType();
+            
+            // Calculate Amount to Credit
+            String prorateType = accountLine.getOverrideProrateType();
+            
+            String account = accountLine.getAccountNumber();
+            String subAccount = accountLine.getSubAccountNumber();
+            String chartCode = accountLine.getChartCode();
+            String objectCode = accountLine.getObjectCode();
+            String projectCode = accountLine.getProjectCode();
+          
+            // To Side:
+
+            // Credit the departmental Account for expense object codes,
+            // or debit the departmental account for revenue object 
+            // codes.
+
+            // Always post the prorated amounts to the account and object
+            // code (no sub-account and sub-object codes) since they are
+            // not guaranteed to exist.
+
+            // Create the origin entry
+            OriginEntryFull creditEntry = new OriginEntryFull(PRJEConstants.ENTRY_DOCTYPE, PRJEConstants.ENTRY_ORIGIN);
+            creditEntry.setFinancialBalanceTypeCode(KFSConstants.BALANCE_TYPE_ACTUAL);
```

Comment:
KFSConstants.BALANCE_TYPE_ACTUAL is deprecated. You should get these values from the system options service. See  edu.arizona.kfs.gl.batch.service.impl.BudgetAdjustmentCashTransferServiceImpl.